### ISSUE TITLE
Enigma failing to decrypt: Decryption failed. Private key not found.

### ISSUE THREAD

**Issue by Incognify** [roundcube/roundcubemail#7839]
I'm able to generate new keys as well as import keys. Recipients also are able to receive emails from my server and decrypt them without error. If I send an email from my roundcube server to another address where I've already imported their public key, it arrives to them encrypted as it should.

The issue is that upon response, server side, it fails to decrypt even if they have the public key of the server side account. HOWEVER, if the key of someone else saved,  it can verify their signature successfully server side.

Any ideas? Looking at the keys folder I can confirm that the keys exist. Encrypted messages just arrive to the inbox like this:

```
Decryption failed. Private key not found. Key ID: B02DB130.
-----BEGIN PGP MESSAGE-----

wcFMA2tZOJytMdLMAQ//X8cKpPKEdN8K/YfE64TYWVD6aWsQyK+aMCQbxkKqJdpq6xWl7GF/ULFY
b3ulm8FFUxOXiDctJM16wMv2zazFCWYq0qyeWoWlBJNgL6mum3RIL3/QI9WkbfJ1sAu1xZfXN6i8
ogT/o9LiDzusswfPCLv0Rs+loQmKrEpELZhOdO0pAJ+xJoubq4Wzd7s+XabWCg7uCOnvGOIckHdN
qunzHorUjcG1NbwpI5PC05yGpAORam9PMPTsXf1oFPEiy7W+qGkgNqinxFMBHsC7uZlskbX4sVdF
```

**Comment by alecpl** [comment]
What do you mean by server-side? I have problem to understand the issue. Please, provide precise steps to reproduce the issue. Would be nice to see email message content, maybe also the keys. Also, enigma_debug log might be useful (of course not one big dump, but split for each operation/step you make).

**Comment by Incognify** [comment]
Ok, I agree. I reread what I wrote and it was not so clear. I created a brand new email account to test this with, and this time it worked without error. Maybe the issue before was it was a pre-existing account on the server? I can try again with the old account, if you would like, but for now it seems resolved.

---

I will label things as "roundcube@domain.com", that is a brand new email address created on the server. The other email is "thunderbird@domain.com" which is my personal email address that I use Thunderbird to manage.

Step 1:

i login to Roundcube ( roundcube@domain.com )

Step 2: I go to Settings > PGP Keys > "Create". I then create a PGP key without error (4096bit)

Success! "A key pair generated and imported successfully."

Step 3: I compose a new email. I want to send my key to "thunderbird@domain.com" so that we may communicate securely. I have, "Attach my public key" selected and send the email. Message sent/received succesfully.

Step 4: In Thunderbird, I see the email from roundcube@domain.com . I respond to it, unencrypted, and share my key for thunderbird@domain.com .   ( I use Enigma in both Thunderbird and in Roundcube. The key from roundcube@domain.com imports successfully into my desktop client.)

Step 5: In Roundcube, I see the email from thunderbird@domain.com and see that it sent me it's public key. So I import it, successfully and without error. (This message contains attached PGP key(s). -> Import)

Step 6: I respond, from Roundcube, to thunderbird@domain.com . Now, both parties have imported the public keys for each other. I send this email with encryption. It's contents are "TESTING TESTING TESTING 1, 2, 3". I enter my PGP password and send the email, encrypted, and signed by roundcube@domain.com

Step 7: It's received without error by thunderbird@domain.com. I can read it's contents, and viewing the source of the email I can see that it's message was fully encrypted in transit. 

I respond to it, and send this message fully encrypted to roundcube@domain.com. Based on everything from above, it "should work."

Step 8: Email is received by roundcube@domain.com and.... It worked. This time.
